### ISSUE TITLE
Build hooks

### ISSUE THREAD

**Issue by sandeep048** [phonegap/build#279]
I would like to find out if the Phonegap build platform allows provision for pre-post build hooks. 
Ref: http://devgirl.org/2013/11/12/three-hooks-your-cordovaphonegap-project-needs/

**Comment by lyptt** [comment]
+1

**Comment by dmytro-shchurov** [comment]
+1

**Comment by PacoBlanco** [comment]
+1

**Comment by cusspvz** [comment]
+1

**Comment by mariano-aguero** [comment]
+1

**Comment by michaelkariv** [comment]
+1

**Comment by speige** [comment]
+1

**Comment by benmcmaster** [comment]
+1

**Comment by escolhatecnologia** [comment]
+1

**Comment by darkdante83** [comment]
+1

**Comment by anemitoff** [comment]
+1

**Comment by tomchiverton** [comment]
Is it at least possible to get a hook to run locally before `phonegap remote` does it stuff ?

**Comment by uerceg** [comment]
+1

**Comment by gabaum10** [comment]
+1, really need this for universal links.

**Comment by victorvhpg** [comment]
+1

**Comment by cusspvz** [comment]
@victorvhpg @uerceg please use github reactions instead.

**Comment by uerceg** [comment]
@cusspvz Why only me and him?

**Comment by cusspvz** [comment]
@uerceg GitHub added recently that functionality, so others didn't knew. Although @gabaum10 has also commented `+1` he explained why he needs.

Its a hassle to receive lots of emails with `+1` comments.

https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments

**Comment by gabaum10** [comment]
Yeah, basically every Cordova plugin for universal links I have found requires hooks to build the necessary entitlements files in iOS.  I don't see another way, unless phonegap build eventually allows us to include additional files like that at the root manually.

**Comment by webshots1** [comment]
It seems like this functionality is required to make iOS universal links work for the Branch.io plugin as well?

Without it I can not get past this error when submitting to iTunes Connect:

ERROR ITMS-90046: "Invalid Code Signing Entitlements. Your application
bundle's signature contains code signing entitlements that are not
supported on iOS. Specifically, value '*' for key
'com.apple.developer.associated-domains' in
'Payload/MY.app/MY' is not supported." 

I believe the solution is to set the ENTITLEMENTS in the xcode file, but I do not think there is any way to modify this in phoengap build...?

**Comment by aaustin** [comment]
Hey @webshots1 - Alex from Branch here. There definitely shouldn't be an issue with a corrupt entitlements file due to the build hooks. It looks like Apple is barfing on a wildcard character `*`. Let's discuss on this issue: https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking/issues/207

Over in that issue, can you post your `<branch-config>` from config.xml?

**Comment by LarsVoith** [comment]
+1

**Comment by Notifium** [comment]
+1

**Comment by nicocedron** [comment]
+1

**Comment by pph7** [comment]
+1

**Comment by mattkenefick** [comment]
+1

**Comment by cusspvz** [comment]
https://github.com/phonegap/build/issues/279#issuecomment-200887687

It's 2018 and there are still `+1` comments :D

**Comment by joshchandler** [comment]
+1

**Comment by ekelokorpi** [comment]
Any update on this?

**Comment by sowobm** [comment]
+1

**Comment by aramgev** [comment]
+1
